repronim/environment/ec2environment.py

```diff
@@ -153,9 +157,93 @@ def execute_command_buffer(self):
         execution.
         """
         host = self._ec2_instance.public_ip_address
-        key_filename = self.get_config('key_filename')
+        key_filename = self['key_filename']
 
         with SSHConnector2(host, key_filename=key_filename) as ssh:
             for command in self._command_buffer:
                 self._lgr.info("Running command '%s'", command['command'])
-                self.execute_command(ssh, command['command'], command['env'])
\ No newline at end of file
+                self.execute_command(ssh, command['command'], command['env'])
+
+    def create_key_pair(self):
+        """
+        Walk the user through creating an SSH key pair that is saved to
+        the AWS platform.
+        """
+
+        def get_user_input(prompt):
+            py3 = version_info[0] > 2
+            if py3:
+                response = input(prompt)
+            else:
+                response = raw_input(prompt)
+            return response.strip()
+
+        # Prompt for a name of the key-pair on AWS.
+        prompt = """
+You did not specify an EC2 SSH key-pair name to use when creating your EC2 environment.
+Please enter a unique name to create a new key-pair or press [enter] to exit.> """
+        key_name = get_user_input(prompt)
+
+        # The user wants to exit.
+        if not key_name:
+            exit(0)
+
+        # Check to see if key_name already exists.
+        for i in range(3):
+            key_pairs = self._ec2_resource.key_pairs.filter(KeyNames=[key_name])
+            try:
+                len(list(key_pairs))
+            except:
+                # Catch the exception raised when there is no matching key name at AWS.
+                break
+            if i == 2:
+                print('That key name exists already, exiting.')
+                exit(1)
+            else:
+                prompt = 'That key name exists already, try again. > '
+                key_name = get_user_input(prompt)
+
+        # Prompt for a path to store the SSH private key.
+        default_path = '{}/.ssh/{}.pem'.format(expanduser("~"), key_name)
+        for i in range(3):
+            prompt = 'Please enter the path to save the private key. [{}]> '.format(default_path)
+            key_filename = get_user_input(prompt)
```

Comment:
I think we shouldn't bother user with this information and just keep those under e.g.

```python
In [9]: os.path.join(AppDirs('repronim', 'repronim.org').user_data_dir, 'ec2_keys')
Out[9]: '/home/yoh/.local/share/repronim/ec2_keys'
```